covid19dp_submission/download_analyses.py

```diff
@@ -67,27 +68,43 @@ def total_analyses_in_project(project):
     return response.json()
 
 
-def get_analyses_to_process(project, num_analyses, total_analyses, processed_analyses_file):
+def get_analyses_to_process(project, num_analyses, total_analyses, processed_analyses_file, ignored_analysis_file, accepted_taxonomies):
     offset = 0
     limit = 100000
-    processed_analyses = get_processed_analyses(processed_analyses_file)
     analyses_for_processing = []
+    analysis_processed = get_analyses_from_file(processed_analyses_file)
+    analysis_to_ignore = get_analyses_from_file(ignored_analysis_file)
+    analysis_to_skip = analysis_processed.union(analysis_to_ignore)
+    new_files_to_ignore = []
     while offset < total_analyses:
-        logger.info(f"Fetching ENA analyses from {offset} to  {offset + limit} (offset={offset}, limit={limit})")
+        logger.debug(f"Fetching ENA analyses from {offset} to  {offset + limit} (offset={offset}, limit={limit})")
         analyses_from_ena = get_analyses_from_ena(project, offset, limit)
-        unprocessed_analyses = filter_out_processed_analyses(analyses_from_ena, processed_analyses)
-        logger.info(
-            f"number of analyses already processed in current iteration: {len(analyses_from_ena) - len(unprocessed_analyses)}")
+        # Filter out based on previously marked analysis
+        unprocessed_analyses = filter_out_processed_analyses(analyses_from_ena, analysis_to_skip)
+        # Filter out based on taxonomy
+        unprocessed_analyses = [a for a in unprocessed_analyses if int(a.get('tax_id') or 0) in accepted_taxonomies]
+        # Gather filtered analysis that had not been previously filtered out
+        new_files_to_ignore.extend([
+            a for a in analyses_from_ena
+            if int(a.get('tax_id') or 0) not in accepted_taxonomies and
+               a.get('analysis_accession') not in analysis_to_ignore
+        ])
```

Comment:
This includes analyses that have already been processed with unacceptable taxonomies, right?  Is that intended?  (I guess it only matters for the first run after this fix is deployed, at which point these all get added to the ignored analysis file anyway.)

Personally I would find it more readable for this code to just split the unprocessed analyses into ones with accepted taxonomies and ones without, up to you though.

Reply:
I've changed this part to make it more readable. It's not perfect but I think it should do for now.
I was checking all the previously processed variant to identified others that might have the wrong taxonomies (and found some). This said, you're right that it should not be done every time.